overlord/snapstate/handlers.go

```diff
@@ -2265,8 +2270,12 @@ func (m *SnapManager) undoLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	// core snap -> next core snap
 	if release.OnClassic && newInfo.Type() == snap.TypeOS && oldCurrent.Unset() {
 		t.Logf("Requested daemon restart (undo classic initial core install)")
-		restart.Request(st, restart.RestartDaemon, nil)
+
+		if err := FinishTaskWithRestart(t, finalStatus, restart.RestartDaemon, nil); err != nil {
+			return err
+		}
 	}
```

Comment:
same here?

Reply:
the if is a big aside, not sure it's better, but I have to rework this code so maybe it will fit somehow differently